src/arch/x86/x86_64/src/pci.rs

```diff
@@ -0,0 +1,37 @@
+use vcell::*;
+
+const MMIO_CFG_BASE: u32 = 0xE000_0000;
+
+pub struct PciAddress {
+    pub segment: u8,  // 4 bits
+    pub bus: u8,      // 8 bits
+    pub device: u8,   // 5 bits
+    pub function: u8, // 3 bits
+    pub offset: u16,  // 12 bits
+}
+
+impl PciAddress {
```

Comment:
is pci properly part of the architecture? Expecially this register below is more part of the cpu or the soc.

Reply:
Moved to SOC (the conservative choice).